src/Browser.php

```diff
@@ -784,7 +880,55 @@ private function requestMayBeStreaming($method, $url, array $headers = array(),
         }
 
         return $this->transaction->send(
-            new Request($method, $url, $headers, $body, $this->protocolVersion)
+            new Request($method, $url, $this->mergeHeaders($headers), $body, $this->protocolVersion)
         );
     }
+
+    /**
+     * @param array $headers
+     * @return array
+     */
+    private function mergeHeaders($headers)
+    {
+        foreach (array_keys($this->defaultHeaders) as $key) {
+            foreach (array_keys($headers) as $headerKey) {
+                if (strcasecmp($headerKey, $key) === 0) {
+                    unset($this->defaultHeaders[$key]);
+                }
+            }
+        }
+
+        return array_merge($this->defaultHeaders, $headers);
+    }
+
+    /**
+     * @param array $headers
+     * @param string $header
+     * @return string|null
+     */
+    private function findCaseInsensitivHeader($headers, $header)
```

Comment:
I see why this was done, but I'd rather inline the method definition to make it easier to follow how they are closely related?

Reply:
got also inlined